viewing_party/party.py

```diff
@@ -1,23 +1,142 @@
 # ------------- WAVE 1 --------------------
 
+
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        return {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+    return None
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    updated_data = user_data.copy()
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            updated_data["watched"].append(movie)
+            updated_data["watchlist"].remove(movie)
+
+    return updated_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
 
+def get_watched_avg_rating(user_data):
+    total_ratings = 0.0
+    total_movies = 0
+    avg_rating = 0.0
+
+    if user_data["watched"]:
+        for movie in user_data["watched"]:
+            total_movies += 1
+            total_ratings += movie["rating"]
+        avg_rating = total_ratings/total_movies
+    return avg_rating
+
+
+def get_most_watched_genre(user_data):
+    most_watched_genre = None
+    genre_count_dict = {}
+    if user_data["watched"]:
+        for movie in user_data["watched"]:
+            if movie["genre"] in genre_count_dict:
+                genre_count_dict[movie["genre"]] += 1
+            else:
+                genre_count_dict[movie["genre"]] = 1
+        counts_list = list(genre_count_dict.values())
+        for genre, count in genre_count_dict.items():
+            if count == max(counts_list):
+                most_watched_genre = genre
+
+    return most_watched_genre

```

Comment:
👍  

I really like how you created a list of all the values in `genre_count_dict` by using `list()` (which also came in handy with `max()`!

My one suggestion is to unnest the core logic of this function by using a guard clause.

```Python
def get_most_watched_genre(user_data):
    most_watched_genre = None
    genre_count_dict = {}
    
    if not user_data["watched"]:
        return None
        
    for movie in user_data["watched"]:
        if movie["genre"] in genre_count_dict:
            genre_count_dict[movie["genre"]] += 1
        else:
            genre_count_dict[movie["genre"]] = 1
    counts_list = list(genre_count_dict.values())
    for genre, count in genre_count_dict.items():
        if count == max(counts_list):
            most_watched_genre = genre

    return most_watched_genre

```